### PR TITLE
feat(events): show all appointments of recurring events

### PR DESCRIPTION
**Pull request by donni106**
This PR introduces a new way of showing events data with considering all appointments of recurring events. Until now we have only respected the current day of an appointment represented by the list date. With the new updates, all appointments result as single events with their own list date to be able to be listed and shown in the calendar on their own.

SVAK-25

| list initially from today | calendar initially from today | calendar next month | event appointments default | event appointments configured |
|---|---|---|---|---|
|Simulator Screen Shot - iPhone 14 Plus - 2024-07-16 at 16 06 05|Simulator Screen Shot - iPhone 14 Plus - 2024-07-16 at 16 05 59|Simulator Screen Shot - iPhone 14 Plus - 2024-07-16 at 16 05 38|Simulator Screen Shot - iPhone 14 Plus - 2024-07-16 at 16 06 15|Simulator Screen Shot - iPhone 14 Plus - 2024-07-16 at 16 08 00|

To test this you will need a local main server set up until https://github.com/smart-village-solutions/smart-village-app-mainserver/pull/431 is merged and deployed.

### CODE REVIEW

**Review by ardasnturk — COMMENTED** [review]
It works well in general. But one thing caught my attention during my tests. I don't know if it's supposed to be exactly like this.

The event on the DetailScreen runs from 07.07.2024 until 29.12.2024 and takes every Wednesday through Friday (also on weekends)

If I go back to the calendar view, this event is not listed for the 14th (Wednesday) and 15th (Thursday) of August. If there is another event on the same day (like on 21.08.2024) then we can see this event.

If there is only one event on that day and it is a serien event, then it is not listed in the calendar view.

|DetailScreen|Calendar View|
|--|--|
Simulator Screenshot - iPhone 15 Pro Max - 2024-07-18 at 10 51 50|Simulator Screenshot - iPhone 15 Pro Max - 2024-07-18 at 10 55 24

File: src/components/Calendar.tsx [hunk 1]
```diff
@@ -33,57 +33,110 @@ type Props = {
...
+
+      return await client.request(
+        getQuery(
+          deprecated?.events?.listingWithoutDateFragment
+            ? QUERY_TYPES.EVENT_RECORDS_WITHOUT_DATE_FRAGMENT
+            : QUERY_TYPES.EVENT_RECORDS
+        ),
+        queryVariablesWithDateRange
+      );
+    },
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/Calendar.tsx:85]: I didn't see any code update but there is a request for review again.

---

Also, the problem I mentioned here continues.
https://github.com/smart-village-solutions/smart-village-app-app/pull/1163#pullrequestreview-2185090655

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: > Also, the problem I mentioned here continues.
https://github.com/smart-village-solutions/smart-village-app-app/pull/1163#pullrequestreview-2185090655

correct, this event is not a recurring event from data point of view. so it cannot be listed multiple times. it is listed with `listDate` logic from the one appointment.
you could try to copy the event with its data and change to a recurring event per cms and check again if it works as you would expect?

[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/Calendar.tsx:85]: I copied an event in CMS. I converted the copied event to a serial event using the same data. All events appear in the list view without any problem. But there are problems when I switch to calendar view.

The first problem I noticed is that even though the event is on 24.07.2024/26.07.2024, I can't see any point in the calendar or click on these dates.

The second problem is that when I click on the date 27.07.2024, as you can see in the screenshot, I can see in the list below not only the events of that day but also the events of the previous days. 


|list view|calendar view|
|--|--|
Simulator Screenshot - iPhone 15 Pro Max - 2024-07-23 at 13 03 06|Simulator Screenshot - iPhone 15 Pro Max - 2024-07-23 at 13 02 59

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: are you on version 4.x.x? there shouldn't be any list below the calendar from the global settings.

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: wow, ok...good catch. i will find out.

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: i found an issue on the main-server an were able to fix it. please have a look again

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: now there are error with some duplicate event and not correctly ordered list by list date :/ i will have to invest more time in the main server part

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: oook, maybe now! puh 🤯

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: nope, online if `dateRange` is given the sorting works. so for the main list it still fails.

[the remark this answers is not in the document]
Comment (donni106) [reply src/components/Calendar.tsx:85]: i am on workarounds, which are nearly working locally...

File: src/components/Calendar.tsx [hunk 2]
```diff
@@ -205,38 +269,39 @@ export const Calendar = ({
...
+        disableArrowLeft={disableArrowLeft}
+        displayLoadingIndicator={loading || isRefetching || isRefetchingSubList}
         firstDay={1}
         markedDates={markedDates}
         markingType="multi-dot"
+        minDate={today}
         onDayPress={onDayPress}
-        renderArrow={renderArrow}
+        onMonthChange={onMonthChange}
+        renderArrow={(direction: Direction) =>
```
Comment (ardasnturk) [inline src/components/Calendar.tsx:296]: I'm sorry but I can tell you that it's still not working correctly. There is no problem with the list view. But the calendar view is still problematic. I will try to explain with screenshots.

I have added a new event. This event runs from 01.07.2024 until 31.08.2024 and repeats only on Tuesdays.  If I go to the Event screen, we can see this in the list. For example, there is one of this event on 30.07.2024. But when I go to the calendar view, I don't see a dot for the event on 30.07.2024 and I can't click on it. Then I click on another date (e.g. 13.08.2024). Then a dot starts to appear under the date 30.07.2024. Then I click on the date 30.07.2024 and I see 2 events, but in the list view only 1 event appears.

As far as I understand, for a recurring event to appear in the calendar, there must be more than one event on that day or the system automatically adds any event to the same day (I am not sure about this).

|||||
|--|--|--|--|
Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 20 30|Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 20 39|Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 20 56|Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 21 03

**Review by ardasnturk — CHANGES_REQUESTED** [review]

File: src/components/Calendar.tsx (hunk 2 above)
Comment (donni106) [reply src/components/Calendar.tsx:296]: i can't reproduce that. as we chatted about this a bit, do you think this is still a problem?

Comment (ardasnturk) [reply src/components/Calendar.tsx:296]: The cause of the problem is just a little confusion. We discussed and resolved this issue on Slack.

**Review by ardasnturk — APPROVED** [review]

**Pull request merged**